packages/web/app/src/lib/preflight-sandbox/graphiql-plugin.tsx

```diff
@@ -165,23 +170,47 @@ export function usePreflightScript(args: {
     'hive:laboratory:isPreflightScriptEnabled',
     false,
   );
-  const [environmentVariables, setEnvironmentVariables] = useLocalStorage(
-    'hive:laboratory:environment',
-    '',
-  );
+
+  // ------------
+  // Result State
+  // ------------
+  //
+  // todo: Probably better to store the result as a single JSON object value.
+  // Use a proper versioned schema with codecs for coercing, defaults, validation, etc.
+  //
+  // todo: Improve `useLocalStorage` by allowing passing a codec? Then we can co-locate
+  // the codec with the data and have it applied transparently, use a decoded value
+  // for the default, etc. ?
+
+  const [environmentVariables, setEnvironmentVariables] = useLocalStorage('hive:laboratory:environment', '{}'); // prettier-ignore
```

Comment:
An empty string will fail JSON parsing. The default value is changed to be something that would parse into an empty state of the value type. Now if JSON parsing fails, it should only be because a stored value was actually invalid. A further improvement might be to notify our error monitoring about it. However given these values also include userland interaction (e.g. user could muck around in their browser local storage) we would need to apply an appropriate filter on incoming errors such as error rate analysis and only keep the most clearly egregious signals for off-work alerting.

Reply:
Turned these thoughts into a todo.